pydantic_ai_slim/pydantic_ai/_tool_manager.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from dataclasses import dataclass, replace
+from typing import Any, Generic
+
+from pydantic import ValidationError
+
+from pydantic_ai.output import DeferredToolCalls
+
+from . import messages as _messages
+from ._run_context import AgentDepsT, RunContext
+from .exceptions import ModelRetry, ToolRetryError, UnexpectedModelBehavior
+from .messages import ToolCallPart
+from .tools import ToolDefinition
+from .toolsets.abstract import AbstractToolset, ToolsetTool
+
+
+@dataclass
+class ToolManager(Generic[AgentDepsT]):
```

Comment:
Something seems unfortunately overly abstract about this name, but maybe it's fine. I don't have a proposal for an alternative.

Reply:
At least it's private :)